sphinx/util/__init__.py

```diff
@@ -130,11 +130,11 @@ def add_file(self, docname, newfile):
             self[newfile][0].add(docname)
             return self[newfile][1]
         uniquename = path.basename(newfile)
-        base, ext = path.splitext(uniquename)
         i = 0
         while uniquename in self._existing:
             i += 1
-            uniquename = '%s%s%s' % (base, i, ext)
+            # Make file name unique while leaving basename intact
+            uniquename = path.join('subdir%s' % i, path.basename(newfile))
```

Comment:
It would be nice if we can keep the original directory structure. But I feel it is not important thing.
@shimizukawa  @shibukawa do you have any opinion?